packages/preset-uno/src/rules/behaviors.ts

```diff
@@ -2,6 +2,60 @@ import { Rule } from '@unocss/core'
 import { handler as h } from '../utils'
 import { colorResolver } from './color'
 
+const outlineStyle = [
+  'auto',
+  'none',
+  'dotted',
+  'dashed',
+  'solid',
+  'double',
+  'groove',
+  'ridge',
+  'inset',
+  'outset',
+  'inherit',
+  'initial',
+  'revert',
+  'unset',
+]
+
+export const outline: Rule[] = [
+  [
+    new RegExp(`^outline-(${outlineStyle.join('|')})$`), ([, d]) => {
+      return {
+        'outline-style': d,
+      }
+    },
+  ],
+  [
+    /^outline-width-(.+)$/, ([, d]) => {
+      const size = h.bracket.fraction.rem(d)
+
+      return {
+        'outline-width': size,
+      }
+    },
+  ],
+  [
+    /^outline-color-(.+)$/, colorResolver('outline-color', 'outline-color'),
+  ],
+  [
+    new RegExp(`^outline-(${outlineStyle.join('|')})-(.+)$`), (reg, config) => {
```

Comment:
I think it would be better to avoid creating dynamic RegExp if possible. We could do the check inside function body instead.